stroyprombeton/context.py

```diff
@@ -204,37 +192,90 @@ def context(self) -> typing.ContextDict:
         }
 
 
-# @todo #449:60m  Remove FetchPositions class.
-#  Create a new context class to search and filter positions.
-#  Move filter_options and slice_products methods to a created class.
-#  FilteredOptions created from `Catalog.filter_positions` is the example.
-class FetchPositions(Catalog):
+class SearchedOptions(context.Context):
```

Comment:
```suggestion
class SearchedOptions(context.ModelContext):
```
we can only rely on this interface, because we implement the public `qs` method here

Reply:
@artemiy312 , the problem here, that will pass the whole context intead of queryset soon.
That's why ModelContext is not appropriate here.

This PR [contains pdd task](https://github.com/fidals/stroyprombeton/pull/516/files#diff-f5058f0d4ecd89be6a462ef5ed432810R165) to improve `**Options` structure. Improving contain base class creation